src/compiler/checker.ts

```diff
@@ -981,6 +982,29 @@ namespace ts {
             }
         }
 
+        function checkAndReportErrorForUsingTypeAsValue(errorLocation: Node, name: string, meaning: SymbolFlags, nameNotFoundMessage: DiagnosticMessage, nameArg: string | Identifier): boolean {
+            const strictlyValueMeanings = SymbolFlags.Value & ~SymbolFlags.Type;

```

Comment:
do not think you need this. we only need to check for either Type or Value, the intersection is not possible since we know we looked for the symbol before and it did not exist.